### DISCUSSION TITLE
Help Request: Flashing Miryoku Layout on "Klein" Keyboard

### DISCUSSION THREAD

**Discussion by 10bn** [manna-harbour/miryoku#323] [Q&A] [answered]
Hi, It’s been quite a while since I last flashed a board, and I’m a bit out of the loop. I’m trying to flash the Miryoku layout onto my “Klein” keyboard which I got second hand, but I’m not having any success. I’d like to use these specific arguments:

` qmk flash -kb klein -km manna-harbour_miryoku -e MIRYOKU_ALPHAS=QWERTY -e MIRYOKU_CLIPBOARD=MAC -e MIRYOKU_NAV=INVERTEDT
`
Klein Repo: https://github.com/snsten/Klein-qmk
The board has 2x ProMicros RP2040, a LED screen on the left side and a rotary encoder on the right side.

Thanks in advance for your support.

**Comment by 10bn** [comment, marked as answer]
I got it settled by copying `/qmk_firmware/layouts/community/split_3x5_3/manna-harbour_miryoku` to `/qmk_firmware/keyboars/klein/keymaps/`

and adding KC_NO to the thumbluster where the klein has space for rotary encoderds.

```
// Copyright 2019 Manna Harbour
// https://github.com/manna-harbour/miryoku

// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 2 of the License, or (at your option) any later version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program. If not, see <http://www.gnu.org/licenses/>.

#pragma once

#define LAYOUT_miryoku(\
K00,   K01,   K02,   K03,   K04,          K05,   K06,   K07,   K08,   K09,\
K10,   K11,   K12,   K13,   K14,          K15,   K16,   K17,   K18,   K19,\
K20,   K21,   K22,   K23,   K24,          K25,   K26,   K27,   K28,   K29,\
N30,   N31,   K32,   K33,   K34,          K35,   K36,   K37,   N38,   N39\
)\
LAYOUT_split_3x5_4(\
K00,   K01,   K02,   K03,   K04,          K05,   K06,   K07,   K08,   K09,\
K10,   K11,   K12,   K13,   K14,          K15,   K16,   K17,   K18,   K19,\
K20,   K21,   K22,   K23,   K24,          K25,   K26,   K27,   K28,   K29,\
       K32,   K33,   K34, KC_NO,          KC_NO, K35,   K36,   K37\
)
```

**Reply by manna-harbour** [reply]
Great!

I'll add a mapping for `split_3x5_4`.

Also, I'm planning on adding the outboards feature from Miryoku ZMK (https://github.com/manna-harbour/miryoku_zmk#outboards) to Miryoku QMK so that keyboard definitions that aren't in a QMK fork (and so can't be built with the workflows merge feature) such as klein can be built from the workflow without needing to copy to a fork.
